### PR TITLE
iio: support of IIO sampling frequency to channel level

### PR DESCRIPTION
**Pull request by yongli3**
Add light/pressure support

Fix issue #2101

Signed-off-by: Yong Li <yong.b.li@intel.com>

### CODE REVIEW

**Comment by yongli3** [comment]
@laykuanloon @edersondisouza @ceolin @fulong82 please help to review/test.

**Comment by yongli3** [comment]
@laykuanloon @edersondisouza @ceolin @fulong82 please help to review/test, in_light/pressure_sampling_frequency is the correct sysfs name?

File: src/modules/flow/iio/nodes.c [hunk 1]
```diff
@@ -904,6 +909,11 @@ light_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_op
 
     mdata->iio_base.config.buffer_size = opts->buffer_size;
     mdata->iio_base.config.sampling_frequency = opts->sampling_frequency;
+    ret = snprintf(mdata->iio_base.config.sampling_frequency_name,
+        sizeof(mdata->iio_base.config.sampling_frequency_name), "%s", "in_light_");
```
Comment (fulong82) [inline src/modules/flow/iio/nodes.c]: This should be `in_illuminance_` and not `in_light_`.
You could refer all iio channel type name here:
http://lxr.free-electrons.com/source/drivers/iio/industrialio-core.c#L54

Comment (yongli3) [inline src/modules/flow/iio/nodes.c]: Thanks @fulong82, done

**Comment by fulong82** [comment]
Tested and it is working. Thanks.

**Comment by ceolin** [comment]
merged, thanks !

**Pull request merged**